java/test/jmri/configurexml/LoadAndStoreTestBase.java

```diff
@@ -94,96 +105,180 @@ public LoadAndStoreTestBase(File file, boolean pass) {
+                if (splits2.length == 2) {  // (yes) remove it
+                    line2 = splits2[0] + splits2[1];
+                }
+            }
+            if (!match && !line1.equals(line2)) {
+                log.error("match failed in LoadAndStoreTest:");
+                log.error("    file1:line {}: \"{}\"", lineNumber1, line1);
+                log.error("    file2:line {}: \"{}\"", lineNumber2, line2);
+                log.error("  comparing file1:\"" + inFile1.getPath() + "\"");
+                log.error("         to file2:\"" + inFile2.getPath() + "\"");
+                Assert.assertEquals(line1, line2);
+            }
+            line1 = next1;
+            line2 = next2;
+        }   // while readLine() != null
 
-    static void loadFile(File inFile) throws Exception {
-        // load file
-        InstanceManager.getDefault(ConfigureManager.class).load(inFile);
+        fileStream1.close();
+        fileStream2.close();
+    }
 
+    // load file
+    public static void loadFile(File inFile) throws Exception {
+        ConfigureManager cm = InstanceManager.getDefault(ConfigureManager.class);
+        boolean good = cm.load(inFile);
+        Assert.assertTrue("loadFile(\"" + inFile.getPath() + "\")", good);
         InstanceManager.getDefault(jmri.LogixManager.class).activateAllLogixs();
         InstanceManager.getDefault(jmri.jmrit.display.layoutEditor.LayoutBlockManager.class).initializeLayoutBlockPaths();
         new jmri.jmrit.catalog.configurexml.DefaultCatalogTreeManagerXml().readCatalogTrees();
     }
 
-    static File storeFile(File inFile) throws Exception {
+    public static File storeFile(File inFile, SaveType inSaveType) throws Exception {
         String name = inFile.getName();
         FileUtil.createDirectory(FileUtil.getUserFilesPath() + "temp");
         File outFile = new File(FileUtil.getUserFilesPath() + "temp/" + name);
-        InstanceManager.getDefault(ConfigureManager.class).storeConfig(outFile);
+
+        ConfigureManager cm = InstanceManager.getDefault(ConfigureManager.class);
+        switch (inSaveType) {
+            case All: {
+                cm.storeAll(outFile);
+                break;
+            }
+            case Config: {
+                cm.storeConfig(outFile);
+                break;
+            }
+            case Prefs: {
+                cm.storePrefs(outFile);
+                break;
+            }
+            case User: {
+                cm.storeUser(outFile);
+                break;
+            }
+            case UserPrefs: {
+                cm.storeUserPrefs(outFile);
+                break;
+            }
+            default: {
+                log.error("Unknown save type {}.", inSaveType);
+                break;
+            }
+        }
+
         return outFile;
     }
 
     @Test
     public void loadLoadStoreFileCheck() throws Exception {
+        if ((saveType == SaveType.All) || (saveType == SaveType.User)) {
```

Comment:
We need to separate the concerns that the test is or is not GUI only from what is being stored (put another way, the assumption that a test that, say, a UserDataManager implementation reads and writes user data correctly can only be run in a GUI is invalid).

I would recommend some parameter like "isGUIOnly" be added to the test parameters and the assumption by changed to ```if (isGUIOnly) { Assume.assumeFalse(GraphicsEnvironment.isHeadless()); }``` and require that the isGUIOnly and StoreType parameters be passed explicitly.

Reply:
Done. Should this also be used to suppress the second load:
```
        // Panel sub-classes (with GUI) will fail if you try to load them twice.
        // (So don't!)
        if ((saveType != SaveType.All) && (saveType != SaveType.User)) {
            loadFile(this.file);
        }
```
```
        if (!guiOnly) {
            loadFile(this.file);
        }
```

Reply:
Or should that be another (class?) parameter also?